src/theme/colour-semantic/theme-helper.ts

```diff
@@ -0,0 +1,144 @@
+import { StyledComponentProps, getCollection, getValue } from "../helpers";
+import { ColourScheme, SemanticColourSet, ThemeCollectionSpec } from "../types";
+import { LifeSGColourSet } from "./specs/lifesg-semantic-tokens";
+import { SemanticColourCollectionMap } from "./types";
+
+const ColourSpec: ThemeCollectionSpec<
+    SemanticColourCollectionMap,
+    ColourScheme
+> = {
+    collections: {
+        lifesg: LifeSGColourSet,
+        bookingsg: LifeSGColourSet,
+        rbs: LifeSGColourSet,
+        mylegacy: LifeSGColourSet,
+        ccube: LifeSGColourSet,
+    },
+    defaultValue: "lifesg",
+};
+
+export const getSemanticColour = (key: keyof SemanticColourSet) => {
+    return (props: StyledComponentProps): string => {
+        const theme = props.theme;
+        const colorSet: SemanticColourSet = getCollection(
+            ColourSpec,
+            theme["colourScheme"]
```

Comment:
dot notation is preferred for stronger typing

```suggestion
            theme.colourScheme
```